warehouse/query-core/src/main/java/datawave/query/planner/FederatedQueryPlanner.java

```diff
@@ -482,79 +479,110 @@ private SortedSet<Pair<Date,Date>> getSubQueryDateRanges(ShardQueryConfiguration
         } else {
             // Otherwise, get the valid date ranges. First, Merge any overlaps.
             SortedSet<Pair<Date,Date>> mergedHoles = mergeRanges(relevantHoles);
+            // Adjust the index holes so that they span from 00:00:00.000 on the first day to 23:59:59.999 on the last day
+            mergedHoles = spanEntireDays(mergedHoles);
             Iterator<Pair<Date,Date>> it = mergedHoles.iterator();
 
-            // If the start of the first hole occurs after the configured start date, add a range spanning from the start date to one day before the start
-            // of the first hole.
+            // If the start of the first index hole occurs after the configured start date, add a range spanning from
+            // the beginDate to one day before the start of the first index hole.
```

Comment:
Should this be "one ms before the start of the first index hole" ?

Reply:
Good catch.  Thanks.